l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -148,103 +98,76 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         uint256 _l2TxGasLimit,
         uint256 _l2TxGasPerPubdataByte
     ) external payable returns (bytes32 l2TxHash) {
-        l2TxHash = deposit(_l2Receiver, _l1Token, _amount, _l2TxGasLimit, _l2TxGasPerPubdataByte, address(0));
+        // we need to deposit and set allowance, as the sender set the allowance to this contract
+        uint256 amount = _depositFundsToSharedBridge(msg.sender, _l1Token, _amount);
```

Comment:
It seems that if we deposit WETH, its ERC20 form is transferred to L2, while the corresponding amount of ETH is still minted on Era. It means that in the edge case when the user tries to withdraw all the deposited ETH, the shared bridge the resources to serve this withdrawal (as the ETH is in the WETH form and I don't see it getting unwrapped anywhere)

Reply:
Weth is unwrapped, (withdrawn) here: 
https://github.com/matter-labs/era-contracts/blob/d4d1654c5cee0873437373a79735e59e1ef994ef/l1-contracts/contracts/bridge/L1SharedBridge.sol#L168

Reply:
I hope answered your question?

Reply:
I see, you moved the logic of unwrapping in the latest commit : )